src/select2.component.ts

```diff
@@ -148,6 +148,29 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
             this.renderer.setElementProperty(this.selector.nativeElement, 'disabled', this.disabled);
         }
     }
+     
+    private setElementValue ( newValue ) {
+        // NOTE:
+        // Multiple values are received from ngOnChanges as a string with single values separated by comma.
+        // Select2OptionData id values should not contain commas for predictable results.
+        // While using commas in ids is a bad idea generally, here may cause malfunctions.
+
+        if (newValue.indexOf(',') != -1) {
```

Comment:
In my latest version I'm checking if the newValue is a string and in that case I'm always splitting by comma.
Doing so it always converts the string to an array. It becomes a single value array if there are no commas.